lib/std/special/compiler_rt/README.md

```diff
@@ -1,15 +1,258 @@
-This compiler-rt library is ported from [LLVM](http://compiler-rt.llvm.org/).
+If hardware lacks basic or specialized functionality, compiler-rt adds such functionality.
+One such example is 64-bit integer multiplication on 32-bit x86.
 
-It's needed because LLVM emits library calls to compiler-rt when hardware lacks
-functionality, for example, 64-bit integer multiplication on 32-bit x86.
+Open question about scope (tracking libgcc):
+- Integer library routines => implemented
+- Soft float library routines => mostly implemented
+- Decimal float library routines => ~120 functions
+- Fixed-point fractional library routines => ~300 functions
+- Exception handling routines => 32 functions not including undocumented ones
+- Miscellaneous routines => cache control and stack function
 
 This library is automatically built as-needed for the compilation target and
 then statically linked and therefore is a transparent dependency for the
 programmer.
+For details see `../compiler_rt.zig`.
 
-Any bugs should be solved by trying to duplicate the bug upstream.
- * If the bug exists upstream, get it fixed with the LLVM team and then port
-   the fix downstream to Zig.
- * If the bug only exists in Zig, something went wrong porting the code,
-   and you can run the C code and Zig code side by side in a debugger
-   to figure out what's happening differently.
+The routines in this folder are listed below.
+Routines are annotated as `type source routine // desription`, with `routine`
+being the name used in aforementioned `compiler_rt.zig`.
+`dev` means deviating from compiler_rt, `port` ported, `source` is the
+information source for the implementation, `none` means unimplemented.
+In the following a brief overview of is given:
+- dev source name_routine, name_routine2 various implementations for performance, simplicity etc
+- port llvm compiler-rt library routines from [LLVM](http://compiler-rt.llvm.org/)
+   * LLVM emits library calls to compiler-rt, if the hardware lacks functionality
+- port softfloat floating point routines from [berkeley-softfloat](https://github.com/ucb-bar/berkeley-softfloat-3)
```

Comment:
Note that currently no functions are ported from berkley softlfoat, the ones I added were adapted from the ones ported from LLVM.